store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/util/DefaultBestEffortsMergeFunction.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.federatedstore.util;
+
+import com.google.common.collect.Lists;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
+import uk.gov.gchq.koryphe.impl.function.IterableConcat;
+import uk.gov.gchq.koryphe.impl.function.ToIterable;
+
+import java.util.function.BiFunction;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
+public class DefaultBestEffortsMergeFunction implements BiFunction<Object, Iterable<Object>, Iterable<Object>> {
+
+    @Override
+    public Iterable<Object> apply(final Object o, final Iterable<Object> objects) {
+        return isNull(objects)
+                ? new ToIterable().apply(o)
+                : new IterableConcat<>().apply(Lists.newArrayList(new ToIterable().apply(o), objects));
```

Comment:
Could this not use the already existing `ToList` koryphe function?

Reply:
Yes it can. Should have noticed this earlier. There's one difference, which is that `ToList().apply(null)` gives a null but `ToIterable().apply(null)` gives an empty Iterable. Refactoring the code here a little can get around this easily enough. I think the new class should **not** be added at all now.
```suggestion
                : new IterableConcat<>().apply(Lists.newArrayList((Iterable<Object>) new ToList().apply(o), objects));
```

Reply:
If you really wanted the `ToList().apply(null)` behaviour to change, you could even add a flag to the Koryphe functionso it does behave like that.